akka-docs/rst/java/stream/stream-integrations.rst

```diff
@@ -7,138 +7,110 @@ Integration
-is delivered to this actor. After that subsequent calls to ``onNext`` will be ignored.
+The sink sends the elements of the stream to the given :class:`ActorRef` that sends back back-pressure signal.
+First element is always `onInitMessage`, then stream is waiting for the given acknowledgement message
+from the given actor which means that it is ready to process elements. It also requires the given acknowledgement
+message after each stream element to make back-pressure work.
 
-You can complete the stream by calling ``onComplete``. After that you are not allowed to
-call ``onNext``, ``onError`` and ``onComplete``.
+If the target actor terminates the stream will be cancelled. When the stream is completed successfully the 
+given ``onCompleteMessage`` will be sent to the destination actor. When the stream is completed with 
+failure a ``akka.actor.Status.Failure`` message will be sent to the destination actor.
 
-You can terminate the stream with failure by calling ``onError``. After that you are not allowed to
-call ``onNext``, ``onError`` and ``onComplete``.
+.. note::
 
-If you suspect that this ``AbstractActorPublisher`` may never get subscribed to, you can override the ``subscriptionTimeout``
-method to provide a timeout after which this Publisher should be considered canceled. The actor will be notified when
-the timeout triggers via an ``ActorPublisherMessage.SubscriptionTimeoutExceeded`` message and MUST then perform
-cleanup and stop itself.
+   Using ``Sink.actorRef`` or ordinary ``tell`` from a ``map`` or ``foreach`` stage means that there is 
+   no back-pressure signal from the destination actor, i.e. if the actor is not consuming the messages 
+   fast enough the mailbox of the actor will grow, unless you use a bounded mailbox with zero 
+   `mailbox-push-timeout-time` or use a rate limiting stage in front. It's often better to 
+   use ``Sink.actorRefWithAck`` or ``ask`` in ``mapAsync``, though. 
 
-If the actor is stopped the stream will be completed, unless it was not already terminated with
-failure, completed or canceled.
+Source.queue
+^^^^^^^^^^^^
 
-More detailed information can be found in the API documentation.
+``Source.queue`` can be used for emitting elements to a stream from an actor (or from anything running outside
+the stream). The elements will be buffered until the stream can process them. You can ``offer`` elements to 
+the queue and they will be emitted to the stream if there is demand from downstream, otherwise they will 
+be buffered until request for demand is received. 
 
-This is how it can be used as input :class:`Source` to a :class:`Flow`:
+Use overflow strategy ``akka.stream.OverflowStrategy.backpressure`` to avoid dropping of elements if the 
+buffer is full.
 
-.. includecode:: ../code/docs/stream/ActorPublisherDocTest.java#actor-publisher-usage
+``SourceQueue.offer`` returns ``CompletionStage<StreamCallbackStatus<Boolean>>`` which completes with ``Success(true)``
```

Comment:
The javadoc seems to be incorrect on those (!) - there is no `StreamCallbackStatus` in the entire project, it seems that `QueueOfferResult` is meant instead?